tests/combine_samples.C

```diff
@@ -91,22 +91,26 @@ int main(int argc, char* argv[])
  
     /*-- Read dimension and count number of snapshots/bases --*/
     if (rank==0) std::cout << "Opening files to read dimension and count number of snapshots/bases" << std::endl;
+    int dimFirst = 0;
     for (const auto& sample_name: sample_names) {
         CAROM::BasisReader reader(sample_name);
 		
         if (kind == "snapshot") {
 	    CAROM::Matrix *snapshots = (CAROM::Matrix*) reader.getSnapshotMatrix(0);
 	    dim    = snapshots->numRows();
  	    snaps += snapshots->numColumns();
+	    if (dimFirst == 0) dimFirst = dim;
         }
 	else {
             CAROM::Matrix *basis = (CAROM::Matrix*) reader.getSpatialBasis(0);
 	    dim    = basis->numRows();
 	    snaps += basis->numColumns();
+	    if (dimFirst == 0) dimFirst = dim;
 	}
     }
 	
     CAROM_VERIFY((snaps > 0) && (dim > 0));
+    CAROM_VERIFY(dim == dimFirst); // help to ensure files have the same dimensions
```

Comment:
If this is inside the loop over `sample_names`, it will verify that all the files are of the same dimension.

Reply:
I see, updated!